gluoncv/data/mhp.py

```diff
@@ -0,0 +1,182 @@
+        assert not np.isnan(np.sum(img_np))
+
+        if self.mode == 'test':
+            img = self._img_transform(img)
+            if self.transform is not None:
+                img = self.transform(img)
+            return img, os.path.basename(self.images[index])
+
+        mask = _get_mask(self.masks[index])
+
+        # algin data
+        w, h = img.size
+        if h < w:
+            oh = 768
+            ow = int(1.0 * w * oh / h + 0.5)
+            if ow % 8:
+                ow = int(round(ow / 8) * 8)
+        else:
+            ow = 768
+            oh = int(1.0 * h * ow / w + 0.5)
+            if oh % 8:
+                oh = int(round(oh / 8) * 8)
+
+        img = img.resize((ow, oh), Image.BILINEAR)
+        mask = mask.resize((ow, oh), Image.NEAREST)
+
+        # synchrosized transform
+        if self.mode == 'train':
+            img, mask = self._sync_transform(img, mask)
+        elif self.mode == 'val':
+            img, mask = self._val_sync_transform(img, mask)
+        else:
+            assert self.mode == 'testval'
+            img, mask = self._img_transform(img), self._mask_transform(mask)
+
+        # general resize, normalize and toTensor
+        if self.transform is not None:
+            img = self.transform(img)
+
+        return img, mask
+
+    def _mask_transform(self, mask):
+        return mx.nd.array(np.array(mask), mx.cpu(0)).astype('int32')   # - 1
+
+    def __len__(self):
+        return len(self.images)
+
+    @property
+    def classes(self):
+        """Category names."""
+        return type(self).CLASSES
+
+    @property
+    def pred_offset(self):
+        return 0
+
+
+def _get_mhp_pairs(folder, split='train'):
+    img_paths = []
+    mask_paths = []
+    img_folder = os.path.join(folder, 'images')
+    mask_folder = os.path.join(folder, 'annotations')
+
+    if split == 'test':
+        img_list = os.path.join(folder, 'test_list.txt')
+    else:
+        img_list = os.path.join(folder, 'train_list.txt')
+
+    with open(img_list) as txt:
+        for filename in txt:
+            # record mask paths
+            mask_short_path = []
+            basename, _ = os.path.splitext(filename)
+            for maskname in os.listdir(mask_folder):
+                if maskname.startswith(basename):
+                    maskpath = os.path.join(mask_folder, maskname)
+                    if os.path.isfile(maskpath):
+                        mask_short_path.append(maskpath)
+                    else:
+                        print('cannot find the mask:', maskpath)
+
+            # mask_short_path is not empty
+            if not mask_short_path:
+                mask_paths.append(mask_short_path)
+
+            # record img paths
+            imgpath = os.path.join(img_folder, filename.rstrip('\n'))
+            if os.path.isfile(imgpath):
+                img_paths.append(imgpath)
+            else:
+                print('cannot find the image:', imgpath)
+
+    if split == 'train':
```

Comment:
Why hardcode these numbers, like 3000? Is this the official data split? If this is, please add some comments or links to explain.

Reply:
Here, we split the dataset to training set, validation set and testing set as the author mentioned on their paper.

Reply:
Got it, thanks.